documentation/src/main/asciidoc/reference/coordination.asciidoc

```diff
@@ -0,0 +1,508 @@
+and thus that `A` changed too.
+====
+
+[[coordination-database-polling]]
+== `database-polling`: additional event tables and polling in background processors
+
+[[coordination-database-polling-basics]]
+=== Basics
+
+The `database-polling` strategy implements coordination through
+<<coordination-database-polling-schema,additional tables>> in the application database.
+
+<<mapper-orm-indexing-automatic,Automatic indexing>> is implemented
+by pushing events to an outbox table within the same transaction as the entity changes,
+and polling this outbox table from background processors which perform indexing.
+
+This strategy is able to provide guarantees that entities will be indexed regardless
+of temporary I/O errors in backend,
+at the cost of being only able to perform this indexing asynchronously.
+
+The `database-polling` strategy can be enabled with the following settings:
+
+[source]
+----
+hibernate.search.coordination.strategy = database-polling
+----
+
+include::components/dynamic-sharding-not-implemented-warning.asciidoc[]
+
+include::components/coordination-comparison-note.asciidoc[]
+
+[[coordination-database-polling-indexing]]
+=== How indexing works with database-polling coordination
+
+[[coordination-database-polling-indexing-detected-changes]]
+Changes have to occur in the ORM session in order to be detected::
+Hibernate Search uses internal events of Hibernate ORM in order to detect changes:
+these events will only be triggered if you actually manipulate managed entity objects in your code,
+updating them by setting their properties or deleting them by calling the appropriate method
+on the Hibernate ORM session.
++
+Conversely, changes resulting from `insert`/`delete`/`update` queries, be it SQL or JPQL/HQL queries,
+are not detected by Hibernate Search.
+This is because queries are executed on the database side,
+without Hibernate having any knowledge of which entities are actually created, deleted or updated.
+One workaround is to explicitly reindex after you run such queries,
+either using the <<mapper-orm-indexing-massindexer,`MassIndexer`>>
+or <<mapper-orm-indexing-manual,manually>>.
+[[coordination-database-polling-indexing-change-filter]]
+Only relevant changes trigger indexing::
+Hibernate Search is aware of the properties that are accessed when building indexed documents.
+Thanks to that knowledge, it is able to skip reindexing
+when a property is modified, but does not affect the indexed document.
++
+You can control this "dirty checking" by setting
+the <<configuration-property-types,boolean property>> `hibernate.search.automatic_indexing.enable_dirty_check`:
++
+* by default, or when set to `true`,
+Hibernate Search will consider whether modified properties are relevant
+before triggering reindexing.
+* when set to `false`, Hibernate Search will trigger reindexing upon any change,
+regardless of the entity properties that changed.
```

Comment:
I would extract these two paragraph to the parent chapter, if these concepts are the same for both configurations.

Reply:
I'd rather keep the first one here, because it's very important and it actually is a limitation of the current coordination strategies; one that we'll hopefully solve with Debezium one day.
I agree about the second one, though; I'll move it somewhere else.

Reply:
After giving it some thought, and considering your other comments: you're right. I'll move everything to their own common section, and leave only links to those sections here (as a reminder).